### PR TITLE
JAT-60 Basic Linter Configuration

### PR DESCRIPTION
**Pull request by selinali2010**
- Setup a simple linter rule
- Configure vscode environment to automatically apply fixes when applicable

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -16,6 +16,13 @@
   "javascript.format.enable": false,
   "typescript.format.enable": false,
 
+  "editor.defaultFormatter": "esbenp.prettier-vscode",
+  "editor.formatOnSave": false,
+  "editor.codeActionsOnSave": {
+    "source.formatDocument": true,
```
Comment (xenown) [inline .vscode/settings.json:22]: Whats the diff between editor.formatOnSave vs source.formatDocument?

File: .eslintrc.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ module.exports = {
     'import/no-unresolved': 'error',
     // Since React 17 and typescript 4.1 you can safely disable the rule
     'react/react-in-jsx-scope': 'off',
+    curly: ['error', 'all'],
```
Comment (xenown) [inline .eslintrc.js:9]: should we put quotes around curly for consistency?

File: .vscode/settings.json (hunk 1 above)
Comment (selinali2010) [reply .vscode/settings.json:22]: `formatOnSave` only applies the default formatter I believe vs using `codeActionsOnSave` which allows us to do multiple actions on save. In this case, `formatDocument` applies prettier fixes and then `fixAll` applies eslint fixes

File: .eslintrc.js (hunk 2 above)
Comment (selinali2010) [reply .eslintrc.js:9]: i did originally, but either prettier or eslint automatically changed it to that lol

**Review by xenown — APPROVED** [review]

**Pull request merged**